src/webapi/rendering_context.rs

```diff
@@ -30,6 +46,201 @@ impl RenderingContext for CanvasRenderingContext2d {
+    /// If the shape has already been closed or has only one point, this function does nothing.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/closePath)
+    pub fn close_path(&self) {
+        js! { @(no_return)
+            @{&self.0}.closePath();
+        }
+    }
+
+    /// Creates a gradient along the line given by the coordinates represented by the parameters.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createLinearGradient)
+    pub fn create_linear_gradient(&self, x0: f64, y0: f64, x1: f64, y1: f64) {
+        //TODO: returns linear gradient
+        js! { @(no_return)
+            @{&self.0}.createLinearGradient(@{x0}, @{y0}, @{x1}, @{y1});
+        }
+    }
+
+    /// Creates a pattern using the specified image (a CanvasImageSource). It repeats the source in 
+    /// the directions specified by the repetition argument. This method returns a CanvasPattern.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern)
+    pub fn create_pattern_image(&self, image: ImageElement, repitition: Option<Repitition>) {
+        let repitition_string;
+        if let Some(repitition) = repitition {
+            match repitition {
+                Repitition::Repeat => {
+                    repitition_string = "repeat";
+                }
+
+                Repitition::RepeatX => {
+                    repitition_string = "repeat-x";
+                }
+
+                Repitition::RepeatY => {
+                    repitition_string = "repeat-y";
+                }
+
+                Repitition::NoRepeat => {
+                    repitition_string = "no-repeat";  
+                }
+            }
+        }
+        else {
+            repitition_string = "repeat";
+        }
+
+        // TODO: returns CanvasPattern
+        js! { @(no_return)
+            @{&self.0}.createPattern(@{image}, @{repitition_string});
+        }
+    }
+
+    /// Creates a radial gradient given by the coordinates of the two circles represented by the parameters. 
+    /// This method returns a CanvasGradient.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createRadialGradient)
+    pub fn create_radial_gradient(&self, x0: f64, y0: f64, r0: f64, x1: f64, y1: f64, r1: f64) {
+        //TODO: returns radial gradient
+        js! { @(no_return)
+            @{&self.0}.createRadialGradient(@{x0}, @{y0}, @{r0}, @{x1}, @{y1}, @{r1});
+        }
+    }
+
+    /// Draws a focus ring around the current path or given path, If a given element is focused.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawFocusIfNeeded)
+    pub fn draw_focus_if_needed(&self, element: HtmlElement) {
```

Comment:
This should probably be generic over `< T: IHtmlElement >` and take `element; &T` to be consistent with the rest of `stdweb`.